src/graphql/getters/getSeasonBadges.ts

```diff
@@ -0,0 +1,295 @@
+      }
+      BADGE_LPOOR {
+        awardedTimestamp
+        mintedTimestamp
+      }
+      BADGE_MONEY_MONEY_MONEY {
+        awardedTimestamp
+        mintedTimestamp
+      }
+      BADGE_WATER_HOSE {
+        awardedTimestamp
+        mintedTimestamp
+      }
+      BADGE_RAIN_MAKER {
+        awardedTimestamp
+        mintedTimestamp
+      }
+      BADGE_DRY_ICE {
+        awardedTimestamp
+        mintedTimestamp
+      }
+    }
+  }`;
+};
+
+const formatTimestamp = (timestamp: number): number | undefined => {
+  if (timestamp === 0) {
+    return undefined;
+  }
+
+  return timestamp * 1000;
+};
+
+type BadgeResponse = {
+  awardedTimestamp: string;
+  mintedTimestamp: string;
+};
+
+type BadgesQueryResponse = {
+  seasonUser: {
+    BADGE_NO_RISK_HERE_SER: BadgeResponse;
+    BADGE_DELTA_DEGEN: BadgeResponse;
+    BADGE_LEVERAGE_CROWBAR: BadgeResponse;
+    BADGE_IRS_CONNOISSEUR: BadgeResponse;
+    BADGE_SUSHI_ROLL: BadgeResponse;
+    BADGE_DEGEN_STUFF: BadgeResponse;
+    BADGE_OK_BOOMER: BadgeResponse;
+    BADGE_LPOOR: BadgeResponse;
+    BADGE_MONEY_MONEY_MONEY: BadgeResponse;
+    BADGE_WATER_HOSE: BadgeResponse;
+    BADGE_RAIN_MAKER: BadgeResponse;
+    BADGE_DRY_ICE: BadgeResponse;
+  };
+};
+
+export type GetProfileBadgesResponse = CollectionBadge[];
+
+function getDefaultResponse(seasonId: Season['id']): GetProfileBadgesResponse {
+  return [
+    {
+      variant: mapForSeason(seasonId, 'fixedTrader'),
+    },
+    {
+      variant: mapForSeason(seasonId, 'deltaDegen'),
+    },
+    {
+      variant: mapForSeason(seasonId, 'leverageCrowbar'),
+    },
+    {
+      variant: mapForSeason(seasonId, 'irsConnoisseur'),
+    },
+    {
+      variant: mapForSeason(seasonId, 'sushiRoll'),
+    },
+    {
+      variant: mapForSeason(seasonId, 'degenStuff'),
+    },
+    {
+      variant: mapForSeason(seasonId, 'topTrader'),
+    },
+    {
+      variant: mapForSeason(seasonId, 'okBoomer'),
+    },
+    {
+      variant: mapForSeason(seasonId, 'maxBidding'),
+    },
+    {
+      variant: mapForSeason(seasonId, 'yikes'),
+    },
+    {
+      variant: mapForSeason(seasonId, 'lpoor'),
+    },
+    {
+      variant: mapForSeason(seasonId, 'moneyMoneyMoney'),
+    },
+    {
+      variant: mapForSeason(seasonId, 'waterHose'),
+    },
+    {
+      variant: mapForSeason(seasonId, 'rainMaker'),
+    },
+    {
+      variant: mapForSeason(seasonId, 'beWaterMyFriend'),
+    },
+    {
+      variant: mapForSeason(seasonId, 'dryIce'),
+    },
+  ];
+}
+
+const mapForSeason = (seasonId: Season['id'], badgeVariant: string): BadgeVariant => {
+  if (seasonId === 0) {
+    return `og${badgeVariant[0].toUpperCase()}${badgeVariant.substring(1)}` as BadgeVariant;
+  }
+  return badgeVariant as BadgeVariant;
+};
+
+export async function getSeasonBadges(
+  owner: string,
+  seasonId: Season['id'],
+): Promise<GetProfileBadgesResponse> {
+  if (!process.env.REACT_APP_SUBGRAPH_BADGES_URL) {
+    return getDefaultResponse(seasonId);
```

Comment:
this is in case something goes wrong with BE... I just return all badges unclaimed and unachieved. Like faking an empty response from the BE to trigger empty state on the FE. Still show decent UI and not get report such as I don't have any badges instead of my screen is white